libs/agno/agno/tools/agentql.py

```
@@ -0,0 +1,112 @@
+from os import getenv
+from typing import Optional
+
+from agno.tools import Toolkit
+from agno.utils.log import logger
+
+try:
+    import agentql
+    from playwright.sync_api import sync_playwright
+except ImportError:
+    raise ImportError("`agentql` not installed. Please install using `pip install agentql`")
+
+
+class AgentQLTools(Toolkit):
+    def __init__(
+        self, api_key: Optional[str] = None, scrape: bool = True, custom_scrape: bool = False, agentql_query: str = ""
+    ):
+        super().__init__(name="agentql_tools")
+
+        self.api_key = api_key or getenv("AGENTQL_API_KEY")
+        if not self.api_key:
+            logger.error("AGENTQL_API_KEY not set. Please set the AGENTQL_API_KEY environment variable.")
+
+        self.agentql_query = agentql_query
+
+        if scrape:
+            self.register(self.scrape_website)
+
+        if custom_scrape:
+            self.register(self.custom_scrape_website)
+
+    def scrape_website(self, url: str) -> str:
+        """
+        Scrape all text content from a website using AgentQL.
+
+        Args:
+            url (str): The URL of the website to scrape
+
+        Returns:
+            str: Extracted text content or error message
+        """
+        if not url:
+            return "No URL provided"
+
+        if not self.api_key:
+            return "No API key provided. Please set the AGENTQL_API_KEY environment variable."
```

Is this necessary? I think we should just raise an exception on init? You can't use the tool without a key, period.

---

We always let the client raise the API key error. That is consistent across all our tools and models